jupyterhub/app.py

```diff
@@ -2193,6 +2192,7 @@ def init_services(self):
                     client_id=service.oauth_client_id,
                     client_secret=service.api_token,
                     redirect_uri=service.oauth_redirect_uri,
+                    allowed_roles=allowed_roles,
```

Comment:
This is already taken care of by the `input=True` trait added to services:

```suggestion
                    allowed_roles=service.allowed_roles,
```

Reply:
Thanks, didn't catch that!